### ISSUE TITLE
Automatically remove products from wishlist after purchase

### ISSUE THREAD

**Issue by Deozaan** [itchio/itch.io#396]
I had a product on my wishlist for a while, then I finally bought it. Some time later I saw it was still on my wishlist.

IMO, things I buy should automatically be removed from my wishlist.

**Comment by fasterthanlime** [comment]
@Deozaan by 'my wishlist' do you mean a collection? I don't think we have an actual wishlisting system

**Comment by Deozaan** [comment]
Yeah. I guess I made a collection named "Wishlist" a long time ago and forgot that it wasn't an actual wishlist. Which explains the behavior of not automatically removing after a purchase.

Sorry. :)

**Comment by leafo** [comment]
I'd prefer to keep collections as the primary way to make lists of things, do you think there should be a collection that works like a wishlist? Another option is to let you follow games, and the ones that you don't own automatically become your wishlist.

**Comment by Deozaan** [comment]
@leafo If you're asking my opinion, I originally opened this thread because I thought that itch.io had a traditional wishlist system. If that were the case, then not removing an item from it after a purchase is something I would consider a bug.

But once I realized it was just my own collection named "Wishlist", I realized it was not a bug. I never actually requested a wishlist feature.

That said, I do like the idea of following games that you don't own becoming a wishlist automatically.

**Comment by stqn1** [comment]
A wishlist feature would be very useful. If I want to buy a game but not now or not at full price, at the moment I only have the collections system to store it somewhere, but then I have to remember to go check that collection regularly…

A wishlist would:
- make it easy to put a game into it (a button next to the "buy" button)
- make it easy to check the wishlist (a button in the top-right)
- send me a mail when the game goes on sale

**Comment by SeanHogan** [comment]
I'd like to chime in that a dedicated wishlist feature would be huge for Itch.io - part of the reason I link to Steam more often is because of how wishlisting on Steam creates two things:

- An (email?) notification to the player when a game becomes purchasable for the first time
- Occasional notifications when the game goes on sale.

**Comment by leafo** [comment]
I plan to make it so that anything you've added to a collection provides those two things, what do you think? (additionally, following a user could trigger similar notifications)

**Comment by SeanHogan** [comment]
Yeah, if you can opt-in to a collection being considered a wishlist. (like I wouldn't want to have pre-existing collections that are just curations, send me e-mails about updates or sales etc).  

Following users could work too, though I'm not sure of the best way to handle notifications there.

**Comment by runevision** [comment]
If users have to do something special like first creating a collection and then choosing that it should work like a wishlist, then the feature won't ever get the same traction as wishlists on other stores.

Devs need to be able to say "Add to your wishlist on itch.io!" and provide a link, and whoever follows the link can then wishlist the game in *one click* once they're on the page.

I have an unreleased game where I'm currently adding a "Add to wishlist on Steam" call to action and link - I was looking for a similar thing for itch.io but have given up since the available functionality doesn't serve the purpose.

**Comment by leafo** [comment]
@runevision I still am planning on treating collections as the wishlist. The UX right now is pretty straight forward, you can click "Add to collection" then confirm right away. You don't need to create a collection before adding something to it since collections are implicitly created.

All collections in your account will be given the wishlist style notifications. (And if that's too much for people, then they can disable notifications on specific collections). So I would recommend telling people to add the game to a collection.

We've already been sending out "A game in your collections is on sale" notifications. I haven't added a method to notify someone when a game page is available yet.

**Comment by runevision** [comment]
Of course I can only respect your choice but I can also say it's nowhere near the usability of e.g. Steam wishlist.

Compare:

Steam wishlist. The word "wishlist" makes the purpose clear from the beginning so the user understands what the interaction is about even before trying to use it. The user can then click the button "Add to your wishlist", which changes it to a checkmark with the text "on your wishlist".

Itch collection: The word "collection" doesn't make it clear what a collection is or what the purpose it. The call to action "add to collection on itch.io" might as well mean to purchase a game for the uninitiated. If someone talks about their collection of games, they usually mean games they own, so the terminology here has a complete mismatch with common language usage. Then, if a user clicks the button, a dialog with multiple choices pops up. You need to scan the whole dialog to infer that 1) there's multiple collections it can be added to, 2) by default it will be added to a collection called backlog. This still doesn't tell me what the implication of the action is. The user might already be scared away from the complexity of the dialog at this point - only if they click the button at the bottom called "Add to collection" will it actually be added.

The itch approach is so much more complex, ambiguous and ultimately confusing than a simple one-click "add to me wishlist" where there's no dialog and the purpose is apparent right in the name of the button itself without already being familiar with it. For this reason I'm just going to skip having a call to action for itch.io since "add to collection on itch" as a call to action simply can't stand on its own without an accompanying explanation of what exactly that means.

**Comment by runevision** [comment]
Let me also say that from a system design standpoint I do understand why collections are neat. They can do the same as wishlists but are more versatile and this is elegant in a way.

However, there's almost always a trade-off between flexibility and simplicity, and the fact is that at the critical point of trying to get a user to take some kind of action to show interest in your unreleased game, any and all complexity in that process will get in the way and lower the chances that the user will follow through with it.

**Comment by Deozaan** [comment]
I think this is a great opportunity for a brand-specific wishlist. Call it an itch-list!

And the button to add/remove something to the wishlist could be similarly themed; it could say something like "I'm itching for this game."

Okay, maybe that's kind of lame. It definitely sounded more clever in my mind before I typed it out. And maybe it only makes sense in English where the phrase "itching for something" or "itching to do something" means craving or wanting (to do) something.

In any case, I agree with @runevision. Collections are great and very versatile, but IMO having them shouldn't preclude having a typical wishlist feature on the site.

**Comment by SeanHogan** [comment]
Gonna agree with runevision here - I love that the collections are versatile, but as far as communicating to players goes, being able to say 'wishlist' goes a long way.

**Comment by merwok** [comment]
I appreciate both viewpoints here!

There’s a compelling argument that it would be best to have something named _Wishlist_ with a clear call to action (button saying _Add to wishlist_) and auto-removal when bought or claimed.

We can also understand that collections are already implemented in the itch codebase and give 90% of what’s needed to add wishlists, so the devs would prefer to reuse that system (I would too!).

So why not have a special collection with a wishlist flag that enables special behaviour!
It would be automatically created when needed, with name automatically set to (translated) _Wishlist_.
When adding a game to the wishlist, there is no dialog to select collection.  (Don’t know if there should be a confirmation or if one-click is best.)
An existing collection would be promoted to wishlist, if there isn’t already one.
The wishlist has all collection functionality: show on profile, delete, rename and un-promote (remove wishlist flag).

**Comment by bkeegan** [comment]
I'm going to second the idea of having a explicit wishlist. I found this page by googling "itch.io wishlist". It's a much clearer "call to action" - we devs are often posting on social media with a call to action to "wishlist our game"

I'd personally like to prioritize itch.io over steam but that's hard to do because there's no real "call to action" I can issue except maybe to follow me on itch.

I know from a software development mentality- it seems redundant, but from a marketing perspective it's not.

**Comment by runevision** [comment]
I added a new issue here:
https://github.com/itchio/itch.io/issues/1218

I propose to use that issue instead of this one to represent this feature request. The issue here (396) was originally about something else, so the first post is not a request to implement an "add to wishlist" button and makes it more cumbersome to get an idea about what the issue is actually about. This also means that this issue is less likely to get 👍 "upvotes" since those are counted based on the first post, but that first posts has text that's about something else. I thought it would be appropriate to have an issue about this feature request where the first post properly represents what it's about, hence the new created issue.
